components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/WADLProcessor.java

```diff
@@ -233,10 +243,17 @@ public String importWADLToRegistry(RequestContext requestContext, String commonL
         registry.put(actualPath, resource);
         addImportAssociations(actualPath);
         if(createService){
-            OMElement serviceElement = getServiceElement(requestContext,
-                    wadlName.contains(".") ? wadlName.substring(0, wadlName.
-                            lastIndexOf(".")) : wadlName, wadlNamespace, actualPath, version);
-            CommonUtil.addService(serviceElement, requestContext);
+	        OMElement serviceElement = RESTServiceUtils.createRestServiceArtifact(wadlElement, wadlName, version,
+	                                                                              RegistryUtils.getRelativePath(
+			                                                                              requestContext
+					                                                                              .getRegistryContext(),
+			                                                                              actualPath));
+            String servicePath = RESTServiceUtils.addServiceToRegistry(requestContext, serviceElement);
+	        addDependency(servicePath, actualPath);
+			String endpointPath = createEndpointElement(requestContext, wadlElement, version);
+	        if(endpointPath != null) {
```

Comment:
@mushthaq33 this code segment is replaced from ed426b50dffaef29768be3f0df8edcd32f5065a1